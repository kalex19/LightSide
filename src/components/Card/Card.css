.card {
    background-color: white;
    border: 1px solid black;
    border-radius: 5px;
    display: flex;
    flex-direction: column;
    font-family: 'Raleway';
    font-size: 14px;
    height: 400px;
    justify-content: space-around;
    margin: auto;
    margin-bottom: 20px;
    opacity: 0.75;
    padding: 10px;
    width: 300px;
}

h2 {
    font-size: 18px;
    font-weight: 500;
}

.headers {
    font-size: 20px;
    font-weight: 900;
}

.favorite {
    background-color: yellow;
}

.favorite-btn {
    margin: 0 20%;
    padding: 2% 7%;
    border: 2px black solid;
    font-size: 1.5em;
}

.favorite-btn:hover {
    background-color: yellow;
}

.no-favorites {
    color: white;
    font-size: 2em;
    margin: 15% 37%;
}

button {
    border: 0px;
    outline: none;
}

.fa-meteor {
    color: orangered;
}