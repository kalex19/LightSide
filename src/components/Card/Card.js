import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';

const Card = (props) => {
    return (
        <section className="card">
            <h2>Name: {props.name}</h2>
            {console.log(props.terrain)}
            <button className='favorited'><span role="img">ðŸ¤©</span></button>
            <h2>{props.birthyear || props.terrain || props.model}</h2>
            <h2>{props.gender || props.diameter || props.vehicle_class}</h2>
            <h2>{props.height || props.population || props.passengers}</h2>
            <h2>{props.eyecolor}</h2> */}
        </section>
    )
}

Card.propTypes = {
    props: PropTypes.string
}

export default Card;