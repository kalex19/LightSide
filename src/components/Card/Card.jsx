import React from 'react';
import './Card.css';
import PropTypes from 'prop-types'; 

const Card = props => {
	const { favoriteCard, info } = props;
	const cardInfo = [];
	for (let key in info) {
		if (key !== 'id' && key !== 'name' && key !== 'favorite') {
			cardInfo.push(
				<p className="card-info">
					{key.toUpperCase()}: {info[key]}
				</p>
			);
		}
	}

	return (
		<section className="card">
			<button onClick={e => favoriteCard(info.id)}>
				<i className={`fab fa-galactic-senate favorite-btn ${info.favorite && 'favorite'}`}> Favorite </i>
			</button>
			<h2>
				<span className="headers"> NAME: </span> {info.name}
			</h2>
			{cardInfo}
		</section>
	);
};

Card.propTypes = {
	info: PropTypes.object,
	key: PropTypes.number,
	favoriteCard: PropTypes.func
};

export default Card;
