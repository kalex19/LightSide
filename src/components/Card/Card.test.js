import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import Card from './Card.jsx';

describe('Card', () => {
    let mockFunc;
    let mockData

    beforeEach (() => {
        mockFunc = jest.fn()
        mockData = {
            name: 'R2D2',
            terrain: 'n/a',
            diameter: '204',
            population: '12999'
        }
    })
    it('should match snapshot', () => {
        const testInfo = {
            class: "starfighter",
            id: "2014-12-10T16:33:52.860000Z",
            model: "Twin Ion Engine/Ln Starfighter",
            name: "TIE/LN starfighter",
            passengers: "0"
        }
        const wrapper = shallow(<Card 
            info={testInfo}
            favoriteCard={jest.fn()}
        />)
        expect(wrapper).toMatchSnapshot()
        
    });

    it('should call favorite card on click', () => {
        const mockCallBack = jest.fn();
        const testInfo = {
            class: "starfighter",
            id: "2014-12-10T16:33:52.860000Z",
            model: "Twin Ion Engine/Ln Starfighter",
            name: "TIE/LN starfighter",
            passengers: "0"
        }
        const wrapper = shallow(<Card 
            info={testInfo}
            favoriteCard={jest.fn()}
        />)

        wrapper.find('button').simulate('click');
        expect(mockCallBack).toBeCalledWith(testInfo.id)

    });

    
})