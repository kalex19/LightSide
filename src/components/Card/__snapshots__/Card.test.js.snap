// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<section
  className="card"
>
  <button
    onClick={[Function]}
  >
    <i
      className="fab fa-galactic-senate favorite-btn undefined"
    >
       Favorite 
    </i>
  </button>
  <h2>
    <span
      className="headers"
    >
       NAME: 
    </span>
     
    TIE/LN starfighter
  </h2>
  <p
    className="card-info"
  >
    CLASS
    : 
    starfighter
  </p>
  <p
    className="card-info"
  >
    MODEL
    : 
    Twin Ion Engine/Ln Starfighter
  </p>
  <p
    className="card-info"
  >
    PASSENGERS
    : 
    0
  </p>
</section>
`;
